### PR TITLE
Add setSync to `Async.Ref`

### PR DESCRIPTION
**Pull request by zainab-ali**
Adds `setSync` and `setSyncPure` methods to `Async.Ref`, as discussed in #923 .

### CODE REVIEW

**Comment by zainab-ali** [comment]
Adding a method to the public `Async.Ref` trait breaks binary compatibility, hence the build failure.

**Comment by mpilquist** [comment]
Yeah, we'll have to add that method to `Task.Ref` only.

File: core/shared/src/main/scala/fs2/Task.scala [hunk 1]
```diff
@@ -342,6 +342,9 @@ object Task extends TaskPlatform with TaskInstances {
     def set(t: Task[A]): Task[Unit] =
       Task.delay { S { t.unsafeRunAsync { r => actor ! Msg.Set(r) } }}
 
+    def setSync(t: Task[A]): Task[Unit] =
+      Task.delay { t.unsafeRunAsync { r => actor ! Msg.Set(r) }}
```
Comment (mpilquist) [inline core/shared/src/main/scala/fs2/Task.scala]: Note that this is still asynchronous -- once the `Msg.Set(r)` message has veen sent to the actor, the resultant `Task[Unit]` is completed. We need to use `Task.async` and pass the callback along to the actor, letting the actor invoke the callback only after the `Set` has happened.

**Comment by mpilquist** [comment]
@zainab-ali Looks good. The 3 remaining errors from MIMA are safe to ignore. Can you add the noted exclusions to build.sbt?

**Comment by zainab-ali** [comment]
As discussed in https://github.com/zainab-ali/fs2-reactive-streams/pull/31, this change is no longer necessary.

**Pull request closed without merging**